utils/generate_types/test/test.ts

```diff
@@ -0,0 +1,499 @@
+  }
+  {
+    const value = await page.mainFrame().evaluate(() => 'hello');
+    const assertion: AssertType<string, typeof value> = true;
+  }
+  {
+    const value = await page.workers()[0].evaluate(() => [1,2,3]);
+    const assertion: AssertType<number[], typeof value> = true;
+  }
+  {
+    const value = await windowHandle.evaluate((x: Window, b) => b, 'world');
+    const assertion: AssertType<string, typeof value> = true;
+  }
+
+
+  {
+    const handle = await page.evaluateHandle(() => 'hello');
+    const value = await handle.jsonValue();
+    const assertion: AssertType<string, typeof value> = true;
+  }
+  {
+    const handle = await page.mainFrame().evaluateHandle(() => ['a', 'b', 'c']);
+    const value = await handle.jsonValue();
+    const assertion: AssertType<string[], typeof value> = true;
+  }
+  {
+    const handle = await page.workers()[0].evaluateHandle(() => 123);
+    const value = await handle.jsonValue();
+    const assertion: AssertType<number, typeof value> = true;
+  }
+  {
+    const handle = await windowHandle.evaluateHandle((x: Window, b) => b, 123);
+    const value = await handle.jsonValue();
+    const assertion: AssertType<number, typeof value> = true;
+  }
+
+  {
+    const handle = await page.evaluateHandle(() => document.createElement('body'));
+    const assertion: AssertType<playwright.ElementHandle<HTMLBodyElement>, typeof handle> = true;
+    await handle.evaluate(body => {
+      const assertion: AssertType<HTMLBodyElement, typeof body> = true;
+    });
+  }
+
+  await browser.close();
+})();
+
+// protocol
+(async () => {
+  const browser = await playwright.chromium.launch();
+  const context = await browser.newContext();
+  const session = await context.createSession(await context.newPage());
+
+
+  session.on('Runtime.executionContextCreated', payload => {
+    const id = payload.context.id;
+    const assertion : AssertType<number, typeof id> = true;
+  })
+
+  const obj = await session.send('Runtime.evaluate', {
+    expression: '1 + 1'
+  });
+  const type = obj.result.type;
+  const assertion : AssertType<string, typeof type> = true;
+  await session.detach()
+
+
+  await browser.close();
+})();
+
+(async () => {
+  const browser = await playwright.firefox.launch();
+  const page = await browser.newPage();
+  const context = page.context();
+  const oneTwoThree = ('pageTarget' in context) ? context['pageTarget'] : 123;
+  const assertion : AssertType<123, typeof oneTwoThree> = true;
+  await browser.close();
+})();
+
+// $eval
+
+(async () => {
+  const browser = await playwright.webkit.launch();
+  const page = await browser.newPage();
+  await page.$eval('span', (element, x) => {
+    const spanAssertion : AssertType<HTMLSpanElement, typeof element> = true;
+    const numberAssertion : AssertType<number, typeof x> = true;
+  }, 5);
+  await page.$eval('my-custom-element', (element, x) => {
+    const elementAssertion : AssertType<Element, typeof element> = true;
```

Comment:
Why Element and not HTMLOrSVGElement? That's exactly the usecase everyone has.

Reply:
Yep, this works. The test was just too lax.